src/vs/platform/backup/common/backup.ts

```diff
@@ -22,10 +23,16 @@ export interface IBackupMainService {
 	 */
 	getWorkspaceBackupPaths(): string[];
 
+	// TODO: Doc
+	getEmptyWorkspaceBackupWindowIds(): string[];
+
 	/**
 	 * Pushes workspace backup paths to be tracked for restoration.
 	 *
 	 * @param workspaces The workspaces to add.
 	 */
 	pushWorkspaceBackupPathsSync(workspaces: Uri[]): void;
+
+	// TODO: Doc
+	pushEmptyWorkspaceBackupWindowIdSync(vscodeWindowId: string): void;
```

Comment:
@Tyriar maybe merge both methods into one and the return type can either be a path to a workspace or a window id? might make the naming easier.